core/tests/protractor_utils/forms.js

```diff
@@ -170,32 +181,36 @@ var ListEditor = function(elem) {
 var RealEditor = function(elem) {
   return {
     setValue: async function(value) {
-      await elem.element(by.tagName('input')).clear();
-      await elem.element(by.tagName('input')).sendKeys(value);
+      var tagNameInput = elem.element(by.tagName('input'));
+      await action.clear('Tag Name Input', tagNameInput);
+      await action.sendKeys('Tag Name Input', tagNameInput, value);
     }
   };
 };
 
 var RichTextEditor = async function(elem) {
   // Set focus in the RTE.
-  await (await elem.all(by.css('.oppia-rte')).first()).click();
+  var oppiaRte = await elem.all(by.css('.oppia-rte')).first();
+  await action.click('Test Oppia Rte', oppiaRte);
 
   var _appendContentText = async function(text) {
-    await (await elem.all(by.css('.oppia-rte')).first()).sendKeys(text);
+    var oppiaRte = await elem.all(by.css('.oppia-rte')).first();
+    await action.sendKeys('Oppia Rte', oppiaRte, text);
   };
   var _clickToolbarButton = async function(buttonName) {
     await waitFor.elementToBeClickable(
-      elem.element(by.css('.' + buttonName)),
-      'Toolbar button takes too long to be clickable.');
-    await elem.element(by.css('.' + buttonName)).click();
+      elem.element(
+        by.css('.' + buttonName)), 'Toolbar button takes too long.');
+    await action.click(
+      'Test Tool Bar Button', elem.element(by.css('.' + buttonName)));
```

Comment:
Now that you're using `action.click`, the `waitFor` above isn't needed. waitFors are included in `action` functions